scripts/zones/Aht_Urhgan_Whitegate/npcs/Shajaf.lua

```diff
@@ -8,13 +8,55 @@ entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    player:startEvent(162)
+    local imperialStanding = player:getCurrency('imperial_standing')
+
+    -- Players can only purchase a KI once every real-life day.
+    -- This var is set onEventFinish
+    if player:getCharVar('isnmAccepted') == 1 then
+        -- 'I don't have any offers for you at the moment'
+        player:startEvent(163)
+    elseif
+        player:hasKeyItem(xi.ki.CONFIDENTIAL_IMPERIAL_ORDER) or
+        player:hasKeyItem(xi.ki.SECRET_IMPERIAL_ORDER)
+    then
+        -- 'What are you doing, running off from a job?'
+        player:startEvent(161, 0)
+    elseif
+        player:hasKeyItem(xi.ki.PSC_WILDCAT_BADGE) and
+        imperialStanding >= 2000 -- Prevent against packet injection
```

Comment:
You dont need to place this check here. The onEventFinish already handles NOT giving the KI if players has cheated his way around the IS.